tests/Simplify/EqualTest.elm

```diff
@@ -934,6 +934,587 @@ a = ({ a = 1 }).a == ({ a = 1 }).b
                             |> Review.Test.whenFixed """module A exposing (..)
 a = 1 == ({ a = 1 }).b
+"""
+                        ]
+        , listTests
+        , setTests
+        , dictTests
+        ]
+
+
+listTests : Test
+listTests =
+    describe "List.length should be List.isEmpty"
+        [ test "should replace List.length l == 0 with List.isEmpty l" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = List.length l == 0
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to List.isEmpty"
+                            , details =
+                                [ "List.length takes as long to run as the list is long"
+                                , "whereas List.isEmpty runs in constant time."
+                                ]
+                            , under = "List.length l == 0"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = List.isEmpty l 
+"""
+                        ]
+        , test "should replace 0 == List.length l with List.isEmpty l" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = 0 == List.length l
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to List.isEmpty"
+                            , details =
+                                [ "List.length takes as long to run as the list is long"
+                                , "whereas List.isEmpty runs in constant time."
+                                ]
+                            , under = "0 == List.length l"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a =  List.isEmpty l
+"""
+                        ]
+        , test "should replace (l |> List.length) == 0 with List.isEmpty l" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = (l |> List.length) == 0
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to List.isEmpty"
+                            , details =
+                                [ "List.length takes as long to run as the list is long"
+                                , "whereas List.isEmpty runs in constant time."
+                                ]
+                            , under = "(l |> List.length) == 0"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = (l |> List.isEmpty) 
```

Comment:
what's the best way to replace this here? I think this should eliminate the pizza and instead replace with `List.isEmpty l` but maybe that's a job for another rule?

Reply:
Yeah that should be another rule. I think this is as it should be :ok_hand: